tests/specs/lemmas.k

```diff
@@ -330,6 +349,23 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ] [ N3 := BUF3:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ] [ N3 := BUF3           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], N, K ) => #padToWidth(K, BUF)
+      requires ( #sizeByteArray( BUF ) <=Int K ) andBool ( M [ N .. K ] ==K #padToWidth( K, .ByteArray ) ) [simplification]
```

Comment:
`M [ N .. K ]` is an expensive calculation to be performing in the side-condition. I wonder if we can avoid it somehow....

The lemma looks fine, but it just seems unfortunate that we have to use it.

Reply:
I was trying to come up with a better solution, and I realized I think the lemma is wrong. When we `#padToWidth` it pads on the left but `#range(_, N, K)` will pad on the right.

Reply:
Sure, if they look correct and keep the tests passing (confirm locally first). Also, maybe best to sync with @PetarMax so you guys are working on different parts of the code.

And finally, make sure to check for duplicate or similar lemmas throughout the codebase! Could be that a small generalization to an existing lemma covers the cases your lemmas are trying to handle. (just general advice, not specific to this situation).